zio-http/src/main/scala/zhttp/http/MediaType.scala

```diff
@@ -0,0 +1,21 @@
+package zhttp.http
+
+case class MediaType(
+  mainType: String,
+  val subType: String,
+  val compressible: Boolean = false,
+  val binary: Boolean = false,
+  val fileExtensions: List[String] = Nil,
+  extensions: Map[String, String] = Map.empty,
+)
+
+object MediaType extends MimeDB {
+  def forExtention(ext: String): Option[MediaType] = extensionMap.get(ext.toLowerCase)
+
+  val extensionMap: Map[String, MediaType] = allMediaTypes.flatMap(m => m.fileExtensions.map(_ -> m)).toMap
+
+  def probeContentType(name: String): Option[String] = name.lastIndexOf(".") match {
```

Comment:
Let's have an unsafe implementation which is package private also. That returns a `null` if nothing matches.

Purely for perf reasons.

Reply:
I've memoised the result for now, will do other perf enhancements in the later PRs.